src/numeric-values.js

```diff
@@ -405,4 +422,405 @@ export function multiplyTypes(type1, type2) {
+      operatorStack.push(token);
+    } else if (token instanceof RightParenthesisToken) {
+      if (!operatorStack.length) {
+        return null;
+      }
+      while (!(last(operatorStack) instanceof LeftParenthesisToken) ) {
+        const o2 = operatorStack.pop();
+        const second = tree.pop();
+        const first = tree.pop();
+        tree.push(toNAryAstNode(o2, first, second));
+      }
+      if (!(last(operatorStack) instanceof LeftParenthesisToken)) {
+        return null;
+      }
+      operatorStack.pop();
+    } else if (token instanceof WhitespaceToken) {
+      // Consume token
+    } else {
+      return null;
+    }
+  }
+  while(operatorStack.length) {
+    if (last(operatorStack) instanceof LeftParenthesisToken) {
+      return null;
+    }
+    const o2 = operatorStack.pop()
+    const second = tree.pop();
+    const first = tree.pop();
+    tree.push(toNAryAstNode(o2, first, second));
+  }
+  return tree[0];
+}
+
+function astToCSSNumericValue(node) {
+  // Recursively transform the expression node into objects, as follows:
+  //
+  if (node.type === 'ADDITION') {
+    // addition node
+    // becomes a new CSSMathSum object, with its values internal slot set to its list of arguments
+    return new CSSMathSum(...node.values.map(value => astToCSSNumericValue(value)));
+  } else if (node.type === 'MULTIPLICATION') {
+    // multiplication node
+    // becomes a new CSSMathProduct object, with its values internal slot set to its list of arguments
+    return new CSSMathProduct(...node.values.map(value => astToCSSNumericValue(value)));
+  } else  if (node.type === 'NEGATE') {
+    // negate node
+    // becomes a new CSSMathNegate object, with its value internal slot set to its argument
+    return new CSSMathNegate(astToCSSNumericValue(node.value));
+  } else if (node.type === 'INVERT') {
+    // invert node
+    // becomes a new CSSMathInvert object, with its value internal slot set to its argument
+    return new CSSMathInvert(astToCSSNumericValue(node.value));
+  } else {
+    // leaf node
+    // reified as appropriate
+    if (node instanceof CSSSimpleBlock) {
+      return reifyMathExpression(new CSSFunction('calc', node.value));
+    } else if (node instanceof IdentToken) {
+      return reifyIdentifier(node);
+    } else {
+      return reifyNumericValue(node);
+    }
+  }
+}
+
+/**
+ * Reify an identifier
+ * https://drafts.css-houdini.org/css-typed-om/#reify-an-identifier
+ * @param {IdentToken} ident
+ */
+function reifyIdentifier(ident) {
+  // Return a new CSSKeywordValue with its value internal slot set to the serialization of ident.
+  return new CSSKeywordValue(ident.value);
+}
+
+/**
+ * Reify a math expression
+ * https://drafts.css-houdini.org/css-typed-om/#reify-a-math-expression
+ * @param {CSSFunction} num
+ */
+function reifyMathExpression(num) {
+  // TODO: handle `clamp()` and possibly other math functions
+  // If num is a min() or max() expression:
```

Comment:
In other files the step numbers of the algorithm was included in the source. I think it would be nice to add these here as well.

```suggestion
  // 1. If num is a min() or max() expression:
```